### ISSUE TITLE
[Suggestions] tips and tricks :)

### ISSUE THREAD

**Issue by decima** [AHAAAAAAA/PokemonGo-Map#1589]
I work a lot with maps and i want to suggest some modifications :)

## Possible Solution
If you use a map only for the rendenring, you can consider using OSM to avoid google api's limitations. 
OSM is full free.
If you want you can also use leafletJS to simplify your usages :)

## Your Environment
* Environment name and version (e.g. Python 2.7): 2.7.12
* Operating System and version (desktop or mobile): Ubuntu 16.04

**Comment by ximex** [comment]
maybe i could help her. i have created many small webmaps with leaflet/openlayers.

**Comment by Schwaitz** [comment]
I would disagree. Google Maps is what should be used, as that is what Pokemon GO uses. The house shapes, roads, paths, etc will be the same.

**Comment by BillyWM** [comment]
How hard is it to get free access to an OSM version of the whole world? I thought you were supposed to download the tiles you need, and any hosted solutions are going to have some kind of access limitation, same as GMaps.

**Comment by BillyWM** [comment]
Open over proprietary is cool and all but the bug tracker isn't getting flooded with reports of people hitting the API limits, so it's a non-issue. I think a painful shift to OSM is a solution in search of a problem.

**Comment by MRokas** [comment]
google api is used for geo-decoding, last time I tried that with OSM it was lame.

**Comment by ximex** [comment]
@MRokas yes the geocoder of OSM (called nominatim) isn't very good.
@BillyWM OSM has the philosophy of "fair use". if you need very hight limits you should set up a own tileserver.
@Schwaitz yes this is right. but if you have the option to choose the layer why not?

What about first only add OSM as an other layer to choose?

**Comment by MRokas** [comment]
@ximex you cant add osm tiles as layer to gmaps. Nor you can put gmap tiles into leaflet

**Comment by ximex** [comment]
@MRokas really? google is a little bit evil ;-)
leaflet could do it with a little hack. but i hard about some performace problems...

**Comment by decima** [comment]
leaflet isn't a bad solution and it's really easy to use, you should have a look at it :) after I don't know what kind of data pokemon Go sends to you, but with leaflet you can add GPS coordinates to put markers on the map :)

**Comment by MRokas** [comment]
@decima - where no real difference which map engine to use. We need GMaps for geodecoding however

**Comment by dorian-marchal** [comment]
Last time I used OSM with Leaflet, it was buggy as hell on Android.
